### PR TITLE
`enterText` to move the caret to the end

### LINKED ISSUE #79494: `TestTextInput.enterText` should put the caret at the end of the text
Currently it sets the caret to (-1, -1): https://master-api.flutter.dev/flutter/flutter_test/TestTextInput/enterText.html.

Typically when the user types the caret stays at the end of the text

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Fixes https://github.com/flutter/flutter/issues/79494. Running TGP to see how breaking this is. [result](https://test.corp.google.com/ui#id=OCL:366163470:BASE:366165211:1617247621441:b5fa92ae)(internal link). 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by LongCatIsLooong** [comment]
Dependent on https://github.com/flutter/flutter/pull/79489

**Comment by skia-gold** [comment]
Gold has detected about 3 untriaged digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/79506

File: packages/flutter/test/material/text_field_test.dart [hunk 1]
```diff
@@ -848,18 +848,24 @@ void main() {
   );
 
   testWidgets('cursor layout has correct width', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController.fromValue(
```
Comment (LongCatIsLooong) [inline packages/flutter/test/material/text_field_test.dart:851]: These golden tests were drawing the caret for position -1.

**Comment by Hixie** [comment]
If you can land this without breaking the world then I am supportive, LGTM. Amusingly it'll conflict with two separate PRs I'm writing right now but luckily it won't do so in a complicated way and those two PRs are already conflicting with each other so whatever. ;-)

**Review by Hixie — APPROVED** [review]
If this requires google3 changes then we should do a migration guide.

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 2]
```diff
@@ -1040,13 +1040,16 @@ class WidgetTester extends WidgetController implements HitTestDispatcher, Ticker
     });
   }
 
-  /// Give the text input widget specified by [finder] the focus and
-  /// enter [text] as if it been provided by the onscreen keyboard.
+  /// Give the text input widget specified by [finder] the focus, replace its
+  /// content with [text], as if it been provided by the onscreen keyboard.
```
Comment (justinmc) [inline packages/flutter_test/lib/src/widget_tester.dart:1044]: Nit:

```diff
- focus, replace its content with [text], as if it been
+ focus and replace its content with [text], as if it had been
```

**Review by justinmc — APPROVED** [review]
LGTM 👍 

I feel like I have run into this before, good call.

**Pull request merged**